lib/log.js

```diff
@@ -12,6 +13,53 @@ const logConfig = config.get('log')
 const StatsDCollector = require('./metrics/statsd')
 const unbuffer = require('./crypto/butil').unbuffer
 
+const userAgent = require('./userAgent')
+const crypto = require('crypto')
+const Amplitude = require('amplitude')
+var amplitude = new Amplitude(config.get('amplitude.apiKey'))
+
+var amplitudeLastSeenTimestamp = 0
+var amplitudeEventCounter = 0
+var amplitudeHmacKey = config.get('amplitude.hmacKey')
+
+function nextAmplitudeEventId(data) {
+  if (data.time > amplitudeLastSeenTimestamp) {
+    amplitudeLastSeenTimestamp = data.time
+    amplitudeEventCounter = 0
+  }
+  return ++amplitudeEventCounter
+}
+
+function anonymizeForAmplitude(id) {
+  return crypto.createHmac('sha256', amplitudeHmacKey)
+               .update(id)
+               .digest('hex')
+}
+
+function sendToAmplitude(data) {
+  if (! data.time) {
+    data.time = Date.now()
+  }
+  userAgent.call(data, data.userAgent, this)
+  amplitude.track({
+    event_type: data.event,
+    event_id: nextAmplitudeEventId(data),
+    device_id: data.flow_id ? anonymizeForAmplitude(data.flow_id) : undefined,
+    user_id: data.uid ? anonymizeForAmplitude(data.uid) : undefined,
+    time: data.time,
+    // Use flowBeginTime as numeric session identifier
+    session_id: data.flow_time ? data.time - data.flow_time : undefined,
+    event_properties: {
+      flow_time: data.flow_time,
+      service: data.service
+    },
+    os_name: data.uaOS,
+    os_version: data.uaOSVersion,
+    platform: data.uaBrowser
```

Comment:
Worth noting: there are fields in this object we can set for Country, City, latitude and longitude.  I'm not comfortable sending the raw lat/long but sending the Country info is probably reasonable, if we can refactor the geo lookup stuff to avoid dong it a second time inside this function.